### PR TITLE
add royalties provider Art Blocks + documentation 

### PR DESCRIPTION
**Pull request by disenotov**

### CODE REVIEW

File: royalties-registry/contracts/providers/RoyaltiesProviderArtBlocks.sol [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            result[2].account = payable(additionalPayee);
+            result[2].value = additionalPart;
+        }
+
+        return result;
+    }
+
+    function setArtblocksAddress(address payable _artblocksAddress) external {
+        require(_artblocksAddress != address(0), "invalid artblocksAddress");
+        require(msg.sender == artblocksAddress, "no permission to change artblocksAddress");
```
Comment (evgenynacu) [inline royalties-registry/contracts/providers/RoyaltiesProviderArtBlocks.sol:62]: I think it's better to skip this check. artblocksAddress can be contract without this. Let's just use Ownable contract, we can then  let artblocks own it

File: royalties-registry/contracts/providers/RoyaltiesProviderArtBlocks.sol [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+import "@rarible/royalties/contracts/IRoyaltiesProvider.sol";
+import "./RoyaltyArtBlocks.sol";
+import "../lib/BpLibrary.sol";
+
+contract RoyaltiesProviderArtBlocks is IRoyaltiesProvider {
+    using SafeMathUpgradeable for uint;
+    using BpLibrary for uint;
+
+    uint96 artblocksPercentage = 250;
```
Comment (evgenynacu) [inline royalties-registry/contracts/providers/RoyaltiesProviderArtBlocks.sol:14]: if this isn't changing, then it should be constant to save gas. But I think it's good to be configurable

**Review by evgenynacu — CHANGES_REQUESTED** [review]

File: royalties-registry/migrations/5_royalties_provider_art_blocks.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+// sets royalties Provider for v2 legacy royalty
+async function setArtBlocksProvider(deployer, royaltiesRegistry, settings){
+    if (!settings.tokens) {
+        return;
+    }
+    if (settings.tokens.length == 0) {
+        return;
+    }
+
+    const artBlocksAddr = (!!settings.artblocksAddress) ? settings.artblocksAddress : ZERO;
```
Comment (evgenynacu) [inline royalties-registry/migrations/5_royalties_provider_art_blocks.js:67]: wil this work? Do you test these migrations?

File: royalties-registry/migrations/5_royalties_provider_art_blocks.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    artblocksAddress: ""
+}
+const mainnet = {
+    tokens: [
+        "0x059edd72cd353df5106d2b9cc5ab83a52287ac3a",
+        "0xa7d8d9ef8d8ce8992df33d8b8cf4aebabd5bd270"
+    ],
+    artblocksAddress: "0x6C093Fe8bc59e1e0cAe2Ec10F0B717D3D182056B"
+}
+const ropsten = {
```
Comment (evgenynacu) [inline royalties-registry/migrations/5_royalties_provider_art_blocks.js:17]: may be there are artblocks on rinkeby or ropsten?

Comment (disenotov) [reply royalties-registry/migrations/5_royalties_provider_art_blocks.js:19]: added rinkeby address

File: royalties-registry/migrations/5_royalties_provider_art_blocks.js (hunk 3 above)
Comment (disenotov) [reply royalties-registry/migrations/5_royalties_provider_art_blocks.js]: yes i tested migartion, it works well. but probably it wouldn't be such a bad idea to add tests for functions used in migrations

File: royalties-registry/contracts/providers/RoyaltiesProviderArtBlocks.sol [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+contract RoyaltiesProviderArtBlocks is IRoyaltiesProvider, OwnableUpgradeable {
+    using SafeMathUpgradeable for uint;
+    using BpLibrary for uint;
+
+    uint96 artblocksPercentage;
+
+    event ArtblocksPercentageChanged(address _who, uint96 _old, uint96 _new);
+
+    function __RoyaltiesProviderArtBlocks_init() external initializer {
+        __Ownable_init();
```
Comment (evgenynacu) [inline royalties-registry/contracts/providers/RoyaltiesProviderArtBlocks.sol:20]: Let's use not upgradeable version of contracts here. no plan to upgrade this, then it's easier to use regular Ownable

**Review by evgenynacu — CHANGES_REQUESTED** [review]

**Pull request merged**